ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,270 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class BarButtonItemView: UIView {
```

Comment:
why are we not inheriting from UIButton like it was before?

Reply:
should the naming be more specific so it doesn't conflate with the iOS control?  like largetitlebarbutton

Reply:
I believe that inheriting from UIButton would not be a good design because it's not just a button instead this view containing two things: UIButton and BadgeView. Also, the same design is already being followed in the case of TabBarItemView.

As we know that iOS naming follows "UI"(UserInterface)  or "NS"(NeXTStep) with their components and aligning to their UIBarButtonItem component which is an item and not just a button because it can take a custom view in its init, so I tried to align the naming very close to theirs so the consumers of this view can easily infer that from the name itself.

Reply:
why isn't the badge view added as a subview of the uibutton? when you press and hold on the bar button I would expect the hovering/press effect to happen in the badge view as well.  tabbaritemview is different because it had to show label vertically bottom to image view and it had different behaviors when rotating.

Reply:
Yeah!! I also realized that adding a tooltip will also make sense if the badge view would be added as a subview to the button because the item's tag is copied to the button's tag. I made this change but it's not working perfectly so I still need to do some fixes. I will update you once it'll be ready.

Reply:
@harrieshin I have now implemented it as you suggested.